### ISSUE TITLE
[SSH] Auto Reveal in Side Bar not working 

### ISSUE THREAD

**Issue by callumdmay** [microsoft/vscode#93973]
Issue Type: <b>Bug</b>

Open an ssh connection to a remote machine, make sure auto reveal ("explorer.autoReveal": true) is set. When the ssh connection is open, the files can be opened but they are not automatically revealed in the side bar.

VS Code version: Code 1.43.1 (fe22a9645b44368865c0ba92e2fb881ff1afce94, 2020-03-18T07:01:20.184Z)
OS version: Windows_NT x64 10.0.18363
Remote OS version: Windows_NT x64 10.0.18363

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-1065G7 CPU @ 1.30GHz (8 x 1498)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: enabled<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.60GB (8.44GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|SSH: *remote*|
|OS|Windows_NT x64 10.0.18363|
|CPUs|Intel(R) Core(TM) i9-9940X CPU @ 3.30GHz (28 x 3312)|
|Memory (System)|63.71GB (39.29GB free)|
|VM|0%|
</details><details><summary>Extensions (7)</summary>

Extension|Author (truncated)|Version
---|---|---
remote-ssh|ms-|0.51.0
remote-ssh-edit|ms-|0.51.0
vscode-remote-extensionpack|ms-|0.20.0
copy-relative-path|ale|0.0.2
cpptools|ms-|0.27.0
save-and-run|wk-|0.0.21
clang-format|xav|1.9.0

</details>

**Comment by roblourens** [comment]
I don't see why this would be different in a remote window. Are you sure you have a folder open? Maybe you can share a screenshot of what you are seeing to make sure I understand?

**Comment by lcherone** [comment]
Same when in container.

![Peek 2020-04-06 15-11](https://user-images.githubusercontent.com/3892806/78567687-f8fb7a00-7818-11ea-9986-d9ef790849d4.gif)

**Comment by roblourens** [comment]
Seems to work for me, @isidorn any ideas?

**Comment by isidorn** [comment]
Whoever is reproducing this can you please:
F1 > developer tools > sources > put a breakpoint [here](https://github.com/microsoft/vscode/blob/9307e0546dc5bd16daedce355769f3767b2d9c59/src/vs/workbench/contrib/files/browser/fileCommands.ts#L292)
Execute the command "Reveal in Side Bar" and try to figure out what is going on

Thanks!

**Comment by PunchyRascal** [comment]
I am experiencing the same thing - connected via SSH to a Linux machine. Clicking on "Reveal in sidebar" does nothing.

For local folder it works fine.

**Comment by dinhngtu** [comment]
I was debugging around explorerView.ts:663:
![image](https://user-images.githubusercontent.com/1257909/79281432-fd780080-7eb2-11ea-80db-45d997738f32.png)
`item.resource` points to the parent directory of the file that's being revealed; at this point, `item.isDisposed` is true and therefore `onSelectResource` is called with an increased `retry`; at `retry === 2` the loop breaks off and no file is revealed. Sometimes `item.resource` doesn't point to the directory containing the file either when the loop exits.
Hope that helps.

**Comment by isidorn** [comment]
@dinhngtu thanks for looking into this and for providing more details.
So it seems like the model is "stale" since those items should not be disposed. I plan to look into this more.
In case you want to continue debugging, here's a code pointer and you can set a condiitional breakpoint [here](https://github.com/microsoft/vscode/blob/ada95d7f3d63fff764d2d1970a7de9151d2a6847/src/vs/workbench/contrib/files/common/explorerModel.ts#L321) to figure out who and why is the parent disposed.

**Comment by gjsjohnmurray** [comment]
#95510 may be related. I have just submitted PR #95555 to fix that.

**Comment by dinhngtu** [comment]
I think I might have figured out how to reproduce the issue.
I tested this using Remote SSH as the only extension.

1. Open any file (e.g. "fs/attr.c");
2. Make sure that Explorer is open;
3. Developer->Reload Window (or close the remote and reconnect);
4. Open another file and reveal it in the side bar; Explorer activates but doesn't point to the correct file.

The issue doesn't happen if either:
- Explorer is not open at the point of the reload;
- or a non-file window is open (e.g. Welcome).

At reload time, if Explorer is open then remoteAgentFileSystemChannel:setCaseSensitive is called, triggering the disposing of all children of the root directory regardless of whether a file is open or not, but the issue only happens when there's an open file.

**Comment by tjanga** [comment]
I can confirm this behavior in WSL 2 as well on 1.45.0-insider

Steps to reproduce:

1. Open a WSL directory
2. Open more than one file
3. Switching among the open files shows no change in the activated selection in explorer.

Note: The selection does change in "Open Editors" appropriately, just not in the actual directory tree.

I can also confirm this does not happen when viewing local (non WSL2) directories.

Lastly, I stumbled upon a workaround:

Clicking on the "reload" icon of the directory tree:
![image](https://user-images.githubusercontent.com/54868496/79709533-421ae780-8277-11ea-9145-d2a1f6d391aa.png)

will correct the behavior for the current session. Upon restart of VSCode, the broken behavior resumes until a manual refresh.

**Comment by isidorn** [comment]
This might be a duplicate of https://github.com/microsoft/vscode/issues/95510
@gjsjohnmurray provided a great PR so this might be fixed from Tuesday's VS Code insiders.

Thus I kindly ask users who see this to try vscode insiders from Tuesday and if they can still reproduce this issue.

@dinhngtu here's a code pointer where the children of the root directy get disposed https://github.com/microsoft/vscode/blob/aecef14e24b4c3643bb6025c75819be4a608be1f/src/vs/workbench/contrib/files/common/explorerService.ts#L63

**Comment by tjanga** [comment]
I installed the latest update available (not sure if it's Tuesday's):

Version: 1.45.0-insider (system setup)
Commit: ec0d19789f89d4c25895f4a0cf364276cfb0c6ed
Date: 2020-04-20T06:08:30.812Z
Electron: 7.2.1
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.19041

And I still observe this behavior. If that's not the right version, could you kindly point me to where I can get it? Thanks.

**Comment by gjsjohnmurray** [comment]
That is Monday's (see the date). Look for it updating tomorrow (Tuesday).

**Comment by tjanga** [comment]
Just installed Tuesday's update:

Version: 1.45.0-insider (system setup)
Commit: b83339a3a7bd8e3ec8a8b79deda9c37e0d91f094
Date: 2020-04-21T05:41:11.873Z
Electron: 7.2.2
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.19041

Problem still exists.

**Comment by isidorn** [comment]
Thanks, assigning to this milestone to investigate further

**Comment by isidorn** [comment]
I have re-wrote how async events are handeld in the explorer view and how we treat disposed elements. So please try this out with Mondays VS Code inisders and let me know if you still see it. Thank you very much.

**Comment by isidorn** [comment]
Did somebody get a change to try this out with latest vs code insiders?

**Comment by tjanga** [comment]
I just downloaded and installed it. I can confirm that this issue has been resolved in the latest update. Thank you!!

I did not have the exact problem described by OP, as mine was showing up in WSL2, so can't speak for OP.

Version: 1.45.0-insider (system setup)
Commit: 0778354f3a6a2c21b7f738a5e6b02e2f1c765e73
Date: 2020-04-27T05:42:11.567Z
Electron: 7.2.2
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.19041
